### ISSUE TITLE
Reading a std::array<Long64_t, 1> from a TFile created on linux does not work on macOS

### ISSUE THREAD

**Issue by ktf** [root-project/root#9001]
To reproduce, download http://ccdb-test.cern.ch:8080/download/d64bd450-1619-11ec-a9d5-200114580202 and open it in TBrowser. Works on ubuntu, but on mac I get:

```
root [2] Error in <TBufferFile::ReadVersion>: Could not find the StreamerInfo with a checksum of 0xdc6299b4 for the class "array<Long64_t,1>" in /Users/ktf/Downloads/array<Long64_t,1>_1631706076679.root.
Error in <TBufferFile::CheckByteCount>: object of class array<Long64_t,1> read too few bytes: 6 instead of 14
```

when clicking on `ccdb_object;1`.

Both ROOT 6.24.02 and 6.24.06 seem to be affected.

**Comment by ktf** [comment]
Notice that using a `std::vector` in place of a `std::array` works as expected.

**Comment by pcanal** [comment]
Does the reading work properly on the same platform (both no warning and the correct value)?  What is the array value for the example provided?

**Comment by ktf** [comment]
yes. Reading it back on linux works just fine.

**Comment by ktf** [comment]
The actual value should be some timestamp around 1630659771715.

**Comment by pcanal** [comment]
:) That looks like a random stream :)

**Comment by ktf** [comment]
yeah, I cannot check the actual value right now, but it looks correct.
